src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -880,7 +880,11 @@ public static bool SequenceEqual<T>(this IEnumerable<T>? first, IEnumerable<T>?
         }
 
         // https://github.com/dotnet/runtime/issues/107723
+#if NET10_0_OR_GREATER
+        public static IEnumerable<T> Reverse<T>(T[] source) => Enumerable.Reverse(source);
```

Comment:
in `net10.0` though won't we get this method for free? Basically it would be coming from the BCL so we wouldn't need it at all.

Reply:
Yes, but the guidelines say to define the non-extension version for binary compatibility, or does that not apply here for some reason?

Reply:
Yes but what idiot wrote those guidelines?  ;) 

I looked back over the guidelines and I'm trying to decide why I added that particular section. I know the reason for creating the non-extension version was to avoid compile time ambiguities. I can't remember exactly why I didn't say "just remove it entirely". I'm guessing it was because there were cases where it was used by full name ... 

Anyways, wrote that for a reason so lets follow it 😄